src/utils/consistency/mod.rs

```diff
@@ -0,0 +1,58 @@
+use self::diff::{Diff, Diffable};
+use crate::config::Config;
+use failure::ResultExt;
+
+mod data;
+mod db;
+mod diff;
+mod index;
+
+pub fn run_check(
+    config: &Config,
+    conn: &postgres::Connection,
+    dry_run: bool,
+) -> Result<(), failure::Error> {
+    if !dry_run {
+        failure::bail!("TODO: only a --dry-run synchronization is supported currently");
+    }
```

Comment:
A quick check against the prod-dump from april (with my index force reset back to then) shows 4159 releases of 1477 crates in the index that are not in the db, and 1482 releases of 1438 crates in the db that are not in the index. There's definitely more work to do before this could be run to fix stuff (mostly around improving queueing behaviour), but I think we can get to a point where it's safe enough to have a non dry-run version of this, and manually fixing the issues seems like it's too large.